router/router_test.go

```diff
@@ -1862,6 +1862,64 @@ var _ = Describe("Router", func() {
 						It("unsuccessfully serves SSL traffic", func() {
 							expectSuccessfulConnection()
 						})
+
+						Context("when verify_client_certificate_metadata is provided ", func() {
+							Context("when the client cert subject is one of the valid subjects for the specified CA subject", func() {
```

Comment:
What about a case where the cert has more info than is specified in the metadata?

For example...
1. The metadata says the country must be USA. 
1. But the cert includes country is USA, and company is Neopets. 

Hopefully it will succeed because the cert matches all of the restrictions explicitly specified in the metadata?

Can we test a case like this?

Reply:
That would actually not match. Internally we convert all of this into a [`pkix.Name`](https://pkg.go.dev/crypto/x509/pkix#Name) as you've likely seen.

The design is for an exact match at the moment, i.e. your configuration needs to be exhaustive.

There is one caveat regarding the matching of `pkix.Name`: We only use the base properties, not `Names` or `ExtraNames` at the moment.

But we can have a test nonetheless that would show exactly that, i.e. that it would not match. Any field that is not defined _must_ be empty right now.